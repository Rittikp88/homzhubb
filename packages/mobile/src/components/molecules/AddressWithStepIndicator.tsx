import React from 'react';
import { FlatList, StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Badge, Label, Text } from '@homzhub/common/src/components';
import { PropertyAddressCountry } from '@homzhub/mobile/src/components/molecules/PropertyAddressCountry';
import { TypeOfPlan } from '@homzhub/common/src/domain/models/AssetPlan';
import { ILabelColor } from '@homzhub/common/src/domain/models/LeaseTransaction';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IProps {
  steps: string[];
  currentIndex: number;
  isStepDone: boolean[];
  primaryAddress: string;
  subAddress: string;
  propertyType: string;
  icon?: string;
  onEditPress?: () => void;
  selectedPan?: string;
  onPressSteps: (index: number) => void;
  badgeStyle?: StyleProp<ViewStyle>;
  stepContainerStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
}

export const AddressWithStepIndicator = (props: IProps): React.ReactElement => {
  const {
    steps,
    currentIndex,
    isStepDone,
    icon,
    selectedPan,
    badgeStyle,
    stepContainerStyle = {},
    containerStyle = {},
    primaryAddress,
    subAddress,
    propertyType,
    onPressSteps,
    onEditPress,
  } = props;

  const { t } = useTranslation(LocaleConstants.namespacesKey.property);

  const { rental, sell, informational, blue, green, darkTint7 } = theme.colors;

  const badgeData = (): ILabelColor => {
    switch (selectedPan) {
      case TypeOfPlan.RENT:
        return {
          label: t('forRental'),
          color: rental,
        };
      case TypeOfPlan.SELL:
        return {
          label: t('forSale'),
          color: sell,
        };
      case TypeOfPlan.MANAGE:
        return {
          label: t('manage'),
          color: informational,
        };
      default:
        return {
          label: t('forRental'),
          color: rental,
        };
    }
  };

  const renderIndicator = ({ item, index }: { item: string; index: number }): React.ReactElement => {
    const isCurrentStep = currentIndex === index;
    const stepDone = isStepDone[index];
    const iconColor = isCurrentStep ? blue : stepDone ? green : darkTint7;
    const labelStyle = [styles.stepLabel, isCurrentStep && styles.currentStepLabel, stepDone && styles.doneStepLabel];
    return (
      <View style={styles.indicatorView}>
        <Icon name={icons.roundFilled} size={30} color={iconColor} onPress={(): void => onPressSteps(index)} />
        <Label type="small" textType="semiBold" style={labelStyle}>
          {item}
        </Label>
      </View>
    );
  };

  const renderSeparator = (separatorData: { highlighted: boolean; leadingItem: string }): React.ReactElement => {
    let isDone = false;
    const indicatorSteps: string[] = steps;
    indicatorSteps.forEach((item, index) => {
      if (separatorData.leadingItem === item) {
        isDone = isStepDone[index];
      }
    });
    return <View style={[styles.separator, isDone && styles.doneSeparator]} />;
  };

  const badge = badgeData();

  return (
    <View style={[styles.container, containerStyle]}>
      <View style={styles.topView}>
        <Text type="small" textType="regular">
          {propertyType}
        </Text>
        {icon && <Icon name={icon} size={23} color={blue} onPress={onEditPress} />}
        {selectedPan && <Badge title={badge.label.toUpperCase()} badgeColor={badge.color} badgeStyle={badgeStyle} />}
      </View>
      <PropertyAddressCountry
        primaryAddress={primaryAddress}
        subAddress={subAddress}
        containerStyle={styles.addressView}
      />
      <FlatList
        keyExtractor={(step): string => step}
        data={steps}
        renderItem={renderIndicator}
        horizontal
        showsHorizontalScrollIndicator={false}
        ItemSeparatorComponent={renderSeparator}
        style={[styles.listStyle, stepContainerStyle]}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
    borderRadius: 4,
    backgroundColor: theme.colors.white,
  },
  addressView: {
    marginVertical: 6,
  },
  indicatorView: {
    alignItems: 'center',
  },
  topView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  separator: {
    backgroundColor: theme.colors.darkTint7,
    width: 50,
    height: 2,
    marginTop: 10,
    opacity: 0.3,
  },
  doneSeparator: {
    backgroundColor: theme.colors.green,
  },
  listStyle: {
    marginTop: 10,
    alignSelf: 'center',
  },
  stepLabel: {
    color: theme.colors.darkTint7,
  },
  currentStepLabel: {
    color: theme.colors.blue,
  },
  doneStepLabel: {
    color: theme.colors.green,
  },
});
