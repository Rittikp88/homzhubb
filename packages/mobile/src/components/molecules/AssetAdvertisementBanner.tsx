import React from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
import { PaginationComponent } from '@homzhub/mobile/src/components/atoms/PaginationComponent';
import { AssetAdvertisement, AssetAdvertisementResults } from '@homzhub/common/src/domain/models/AssetAdvertisement';

interface IAssetAdvertisementBannerState {
  banners: AssetAdvertisement;
  activeSlide: number;
}

export class AssetAdvertisementBanner extends React.PureComponent<{}, IAssetAdvertisementBannerState> {
  public state = {
    banners: {} as AssetAdvertisement,
    activeSlide: 0,
  };

  public componentDidMount = async (): Promise<void> => {
    await this.fetchAdvertisementBanners();
  };

  public render(): React.ReactNode {
    const { banners, activeSlide } = this.state;
    if (!banners) {
      return null;
    }
    return (
      <View style={styles.container}>
        <SnapCarousel
          carouselData={banners?.results ?? []}
          carouselItem={this.renderCarouselItem}
          itemWidth={theme.viewport.width}
          activeIndex={activeSlide}
          onSnapToItem={this.onSnapToItem}
          testID="bannerSnap"
        />
        <PaginationComponent
          dotsLength={banners?.results?.length ?? 0}
          activeSlide={activeSlide}
          containerStyle={styles.overlay}
          activeDotStyle={styles.activeDotStyle}
          inactiveDotStyle={styles.inactiveDotStyle}
        />
      </View>
    );
  }

  private renderCarouselItem = (item: AssetAdvertisementResults): React.ReactElement => {
    return (
      <Image
        source={{
          uri: item.attachment.link,
        }}
        style={styles.carouselImage}
      />
    );
  };

  public onSnapToItem = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };

  public fetchAdvertisementBanners = async (): Promise<void> => {
    const response: AssetAdvertisement = await DashboardRepository.getAdvertisements();
    this.setState({ banners: response });
  };
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 4,
    height: 180,
    overflow: 'hidden',
  },
  overlay: {
    position: 'absolute',
    flex: 0,
    alignSelf: 'center',
    bottom: 0,
    paddingVertical: 5,
  },
  carouselImage: {
    height: '100%',
    width: '100%',
  },
  activeDotStyle: {
    borderWidth: 1,
  },
  inactiveDotStyle: {
    backgroundColor: theme.colors.darkTint6,
    borderColor: theme.colors.transparent,
  },
});
