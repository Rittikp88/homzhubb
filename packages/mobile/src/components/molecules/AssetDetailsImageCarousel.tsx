import React from 'react';
import { View, StyleSheet, Image, TouchableOpacity } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { ImageVideoPagination } from '@homzhub/common/src/components';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';

interface IProps {
  images: any;
  enterFullScreen: () => void;
  updateSlide: (index: number) => void;
  activeSlide: number;
}

export class AssetDetailsImageCarousel extends React.PureComponent<IProps> {
  public render(): React.ReactElement {
    const { activeSlide, images } = this.props;
    return (
      <View style={styles.carouselContainer}>
        <SnapCarousel
          carouselData={images}
          carouselItem={this.renderCarouselItem}
          activeIndex={activeSlide}
          sliderWidth={theme.viewport.width}
          itemWidth={theme.viewport.width}
          onSnapToItem={this.onSnapToItem}
        />
        <View style={styles.overlay}>
          <ImageVideoPagination
            currentSlide={activeSlide}
            totalSlides={images.length}
            type={images[activeSlide].attachment_type}
          />
        </View>
      </View>
    );
  }

  private renderCarouselItem = (item: any): React.ReactElement => {
    const { enterFullScreen } = this.props;
    return (
      <TouchableOpacity onPress={enterFullScreen}>
        {item.attachment_type === 'IMAGE' && (
          <Image
            source={{
              uri: item.link,
            }}
            style={styles.carouselImage}
          />
        )}
        {/* TODO: Remove once the api is integrated */}
        {item.attachment_type === 'VIDEO' && <Image source={item.thumbnail_url} style={styles.carouselImage} />}
      </TouchableOpacity>
    );
  };

  public onSnapToItem = (slideNumber: number): void => {
    const { updateSlide } = this.props;
    updateSlide(slideNumber);
  };
}

const styles = StyleSheet.create({
  carouselContainer: {
    borderRadius: 4,
    overflow: 'hidden',
    backgroundColor: theme.colors.darkTint1,
  },
  overlay: {
    position: 'absolute',
    alignSelf: 'center',
    bottom: 15,
    backgroundColor: theme.colors.carouselCardOpacity,
  },
  carouselImage: {
    height: '100%',
    width: '100%',
  },
});
