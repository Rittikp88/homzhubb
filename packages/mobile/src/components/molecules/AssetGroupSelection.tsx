import React from 'react';
import { View, StyleSheet, TouchableOpacity, ScrollView } from 'react-native';
import { WithTranslation, withTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon from '@homzhub/common/src/assets/icon';
import { Label, Text } from '@homzhub/common/src/components';
import { AssetGroup } from '@homzhub/common/src/domain/models/AssetGroup';
import { Unit } from '@homzhub/common/src/domain/models/Unit';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IProps extends WithTranslation {
  assetGroups: AssetGroup[];
  selectedAssetGroupType: number;
  selectedAssetGroupId: number;
  scrollRef: ScrollView | null;
  onAssetGroupSelected: (id: number) => void;
  isDisabled: boolean;
}

interface IOwnState {
  selectedAssetGroup: number;
}

enum Type {
  // eslint-disable-next-line no-shadow
  AssetGroup = 'AssetGroup',
  AssetType = 'AssetType',
}

class AssetGroupSelection extends React.PureComponent<IProps, IOwnState> {
  public state = {
    selectedAssetGroup: -1,
  };

  public static getDerivedStateFromProps(props: IProps, state: IOwnState): IOwnState | null {
    const { selectedAssetGroupId: newPropValue } = props;
    const { selectedAssetGroup: oldStateValue } = state;

    if (oldStateValue !== newPropValue && newPropValue !== -1) {
      return {
        selectedAssetGroup: newPropValue,
      };
    }

    return null;
  }

  public render = (): React.ReactNode => {
    const { assetGroups, t, isDisabled } = this.props;
    const { selectedAssetGroup } = this.state;

    const selectedGroup = assetGroups.find((assetGroup: AssetGroup) => assetGroup.id === selectedAssetGroup);

    let opacity;
    let pointer;
    if (isDisabled) {
      opacity = 0.5;
      pointer = 'none';
    }

    return (
      <View style={[styles.container, { opacity }]} pointerEvents={pointer as 'none' | undefined}>
        <Text type="small" textType="semiBold" style={styles.title}>
          {t('myProjectIs')}
        </Text>
        <View style={styles.mainGroupContainer}>
          {assetGroups.map((group, index) => this.renderItem(Type.AssetGroup, group, index))}
        </View>
        {selectedAssetGroup >= 0 && (
          <>
            <Text type="small" textType="semiBold" style={styles.title}>
              {t('selectType')}
            </Text>
            <View style={styles.subGroupContainer}>
              {selectedGroup?.assetTypes.map((item, index) => this.renderItem(Type.AssetType, item, index))}
            </View>
          </>
        )}
      </View>
    );
  };

  private renderItem = (type: Type, item: AssetGroup | Unit, index: number): React.ReactNode => {
    const { selectedAssetGroup } = this.state;
    const { onAssetGroupSelected, selectedAssetGroupType, scrollRef } = this.props;

    const conditionalSelectedItem = type === Type.AssetGroup ? selectedAssetGroup : selectedAssetGroupType;
    const conditionalContainerStyle =
      type === Type.AssetGroup
        ? styles.itemContainer
        : [styles.subItemContainer, { marginEnd: index % 2 === 0 ? marginEnd : 0 }];

    let color = theme.colors.darkTint4;
    let backgroundColor;
    let textType = 'regular';
    if (item.id === conditionalSelectedItem) {
      color = theme.colors.white;
      backgroundColor = theme.colors.primaryColor;
      textType = 'semiBold';
    }

    const onPress = (): void => {
      if (type === Type.AssetGroup) {
        this.setState({ selectedAssetGroup: item.id });
        onAssetGroupSelected(-1);
        setTimeout(() => {
          scrollRef?.scrollToEnd({ animated: true });
        }, 0);
      } else {
        onAssetGroupSelected(item.id);
      }
    };

    return (
      <TouchableOpacity key={item.id} onPress={onPress} style={[conditionalContainerStyle, { backgroundColor }]}>
        {type === Type.AssetGroup && <Icon name={item.name.toLowerCase()} size={24} color={color} />}
        <Label type="large" textType={textType as 'regular' | 'semiBold'} style={[styles.assetGroupName, { color }]}>
          {item.name}
        </Label>
      </TouchableOpacity>
    );
  };
}

const {
  layout: { screenPadding },
  viewport: { width },
} = theme;
const marginEnd = 12;

const styles = StyleSheet.create({
  container: {
    marginHorizontal: screenPadding,
    marginBottom: 12,
  },
  mainGroupContainer: {
    flexDirection: 'row',
  },
  subGroupContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  itemContainer: {
    alignItems: 'center',
    marginEnd: 12,
    paddingVertical: 8,
    paddingHorizontal: 16,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: theme.colors.disabled,
  },
  subItemContainer: {
    width: (width - (screenPadding * 2 + marginEnd)) / 2,
    paddingVertical: 4,
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 12,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: theme.colors.disabled,
  },
  assetGroupName: {
    textAlign: 'center',
    marginEnd: 4,
  },
  title: {
    textAlign: 'center',
    marginTop: 24,
    marginBottom: 16,
  },
});

const HOC = withTranslation(LocaleConstants.namespacesKey.property)(AssetGroupSelection);
export { HOC as AssetGroupSelection };
