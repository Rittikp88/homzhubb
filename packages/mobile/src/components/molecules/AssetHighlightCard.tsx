import React, { Component } from 'react';
import { StyleSheet, FlatList, TouchableOpacity } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { Label } from '@homzhub/common/src/components/atoms/Text';
import { SVGUri } from '@homzhub/common/src/components/atoms/Svg';
import { AssetListingSection, PaginationComponent, SnapCarousel } from '@homzhub/mobile/src/components';
import { Amenity } from '@homzhub/common/src/domain/models/Amenity';

interface IProps {
  title: string;
  data: Amenity[];
  selectedAmenity: number[];
  onAmenityPress: (id: number) => void;
}

interface IState {
  activeSlide: number;
}

class AssetHighlightCard extends Component<IProps, IState> {
  public state = {
    activeSlide: 0,
  };

  public render(): React.ReactNode {
    const { title } = this.props;
    const { activeSlide } = this.state;
    const formattedData = this.getFormattedData();

    return (
      <AssetListingSection title={title} containerStyles={styles.container}>
        <>
          <SnapCarousel
            carouselData={formattedData}
            carouselItem={this.carouselItem}
            activeIndex={activeSlide}
            onSnapToItem={this.onSnapToItem}
            containerStyle={styles.carouselContainer}
          />
          <PaginationComponent
            containerStyle={styles.pagination}
            dotsLength={formattedData.length}
            activeSlide={activeSlide}
            activeDotStyle={styles.activeDot}
            inactiveDotStyle={styles.inactiveDot}
          />
        </>
      </AssetListingSection>
    );
  }

  private renderListItem = ({ item }: { item: Amenity }): React.ReactElement => {
    const { selectedAmenity, onAmenityPress } = this.props;
    const isSelected = selectedAmenity.includes(item.id);
    return (
      <TouchableOpacity style={styles.amenityItem} onPress={(): void => onAmenityPress(item.id)}>
        <SVGUri
          uri={item.attachment.link}
          height={30}
          width={30}
          stroke={isSelected ? theme.colors.active : undefined}
          strokeWidth={isSelected ? 0.5 : undefined}
        />
        <Label type="regular" textType="regular" style={isSelected && { color: theme.colors.blue }}>
          {item.name}
        </Label>
      </TouchableOpacity>
    );
  };

  private onSnapToItem = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };

  private getFormattedData = (): Amenity[][] => {
    const { data } = this.props;
    let index;
    const arrayLength = data.length;
    const tempArray = [];
    let chunk = [];
    for (index = 0; index < arrayLength; index += 9) {
      chunk = data.slice(index, index + 9);
      tempArray.push(chunk);
    }
    return tempArray;
  };

  private carouselItem = (item: Amenity[]): React.ReactElement => {
    return (
      <FlatList
        data={item}
        numColumns={3}
        renderItem={this.renderListItem}
        contentContainerStyle={styles.listContainer}
      />
    );
  };
}

export default AssetHighlightCard;

const styles = StyleSheet.create({
  container: {
    marginBottom: 20,
  },
  carouselContainer: {
    alignSelf: 'center',
  },
  listContainer: {
    paddingTop: 16,
  },
  pagination: {
    paddingVertical: 0,
  },
  amenityItem: {
    width: (theme.viewport.width - 32) / 3,
    alignItems: 'center',
    marginBottom: 16,
  },
  activeDot: {
    borderWidth: 1,
  },
  inactiveDot: {
    backgroundColor: theme.colors.darkTint10,
    borderWidth: 0,
  },
});
