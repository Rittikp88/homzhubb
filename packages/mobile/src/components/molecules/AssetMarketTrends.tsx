import React from 'react';
import { Image, StyleSheet, TouchableOpacity, View } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/mobile/src/utils/AlertHelper';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { LinkingService } from '@homzhub/mobile/src/services/LinkingService';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { ImagePlaceholder } from '@homzhub/common/src/components/atoms/ImagePlaceholder';
import { Text, Label } from '@homzhub/common/src/components/atoms/Text';
import { MarketTrends } from '@homzhub/common/src/domain/models/MarketTrends';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IProps extends WithTranslation {
  isHeaderVisible?: boolean;
  onViewAll?: () => void;
  limit?: number;
}

interface IMarketTrendsState {
  data: MarketTrends;
}

export class AssetMarketTrends extends React.PureComponent<IProps, IMarketTrendsState> {
  public state = {
    data: {} as MarketTrends,
  };

  public componentDidMount = async (): Promise<void> => {
    await this.getMarketTrends();
  };

  public render(): React.ReactNode {
    const { onViewAll, t, isHeaderVisible = true } = this.props;
    const { data } = this.state;
    return (
      <View style={styles.container}>
        {isHeaderVisible && (
          <View style={styles.header}>
            <Icon name={icons.increase} color={theme.colors.darkTint4} size={20} />
            <Text type="regular" textType="semiBold" style={styles.headerText}>
              {t('marketTrends')}
            </Text>
            {data.results?.length > 0 && (
              <Text type="small" textType="semiBold" onPress={onViewAll} style={styles.viewAll}>
                {t('viewAll')}
              </Text>
            )}
          </View>
        )}
        {this.renderTrends()}
      </View>
    );
  }

  public renderTrends = (): React.ReactNode => {
    const {
      data: { results },
    } = this.state;
    const { t } = this.props;

    if (results?.length === 0) {
      return (
        <View style={styles.noData}>
          <Text type="small" textType="regular">
            {t('noTrends')}
          </Text>
        </View>
      );
    }

    return (results || []).map((item) => {
      const { title, postedAtDate, link, attachment } = item;
      const onLinkPress = async (): Promise<void> => {
        await LinkingService.canOpenURL(link);
      };

      return (
        <TouchableOpacity key={`${item.id}`} onPress={onLinkPress} style={styles.trendContainer} testID="linkTouch">
          {attachment && !!attachment.link ? (
            <Image source={{ uri: attachment?.link }} style={styles.image} />
          ) : (
            <ImagePlaceholder height={80} width={80} containerStyle={styles.placeHolderImage} />
          )}
          <View style={styles.trendValuesContainer}>
            <Text type="small" textType="regular" style={styles.trendHeader} numberOfLines={2}>
              {title}
            </Text>
            <View style={styles.trendData}>
              <Label type="large" textType="regular" style={styles.trendDate}>
                {postedAtDate}
              </Label>
              <Icon name={icons.dart} color={theme.colors.primaryColor} size={18} />
            </View>
          </View>
        </TouchableOpacity>
      );
    });
  };

  public getMarketTrends = async (): Promise<void> => {
    const { limit = 2 } = this.props;
    try {
      const response: MarketTrends = await DashboardRepository.getMarketTrends(limit);
      this.setState({ data: response });
    } catch (err) {
      AlertHelper.error({ message: ErrorUtils.getErrorMessage(err.details) });
    }
  };
}

export default withTranslation(LocaleConstants.namespacesKey.assetDashboard)(AssetMarketTrends);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 26,
    paddingBottom: 8,
    paddingHorizontal: 16,
    backgroundColor: theme.colors.white,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 16,
  },
  headerText: {
    marginStart: 12,
    flex: 1,
  },
  trendContainer: {
    flexDirection: 'row',
    marginBottom: 16,
    padding: 12,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: theme.colors.background,
  },
  viewAll: {
    color: theme.colors.primaryColor,
  },
  trendDate: {
    color: theme.colors.darkTint5,
  },
  trendValuesContainer: {
    flex: 1,
    justifyContent: 'space-between',
    marginStart: 12,
  },
  trendData: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  trendHeader: {
    color: theme.colors.darkTint2,
    flexWrap: 'wrap',
  },
  noData: {
    flex: 1,
    minHeight: 100,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: 80,
    height: 80,
    borderRadius: 4,
  },
  placeHolderImage: {
    borderColor: theme.colors.darkTint10,
    borderWidth: 1,
    borderRadius: 4,
  },
});
