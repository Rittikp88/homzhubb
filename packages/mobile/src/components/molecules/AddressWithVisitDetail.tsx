import React from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { DateUtils } from '@homzhub/common/src/utils/DateUtils';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Label, Text } from '@homzhub/common/src/components/atoms/Text';
import { PropertyAddress } from '@homzhub/common/src/components/molecules/PropertyAddress';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { VisitSlot } from '@homzhub/common/src/mocks/BookVisit';

interface IProps {
  primaryAddress: string;
  subAddress: string;
  startDate: string;
  endDate: string;
  comments?: string;
  isMissedVisit?: boolean;
  isCompletedVisit?: boolean;
  isRescheduleAll?: boolean;
  isFromProperty?: boolean;
  onPressSchedule?: () => void;
  containerStyle?: StyleProp<ViewStyle>;
}

export const AddressWithVisitDetail = (props: IProps): React.ReactElement => {
  const {
    subAddress,
    primaryAddress,
    startDate,
    comments,
    isMissedVisit,
    isCompletedVisit,
    containerStyle = {},
    onPressSchedule,
    isRescheduleAll = false,
    isFromProperty = false,
  } = props;
  const { t } = useTranslation(LocaleConstants.namespacesKey.property);
  const dateTime = DateUtils.convertTimeFormat(startDate, 'YYYY-MM-DD HH');
  const time = VisitSlot.find((item) => item.from === Number(dateTime[1]));
  const textStyle = [styles.textColor, isMissedVisit && styles.missedColor];
  return (
    <View style={containerStyle}>
      {isFromProperty ? (
        <>
          {!!comments && (
            <Text type="small" textType="semiBold" style={styles.addressContainer}>
              {comments}
            </Text>
          )}
        </>
      ) : (
        <PropertyAddress
          primaryTextType="small"
          subAddress={subAddress}
          primaryAddress={primaryAddress}
          subAddressStyle={styles.subAddress}
          containerStyle={styles.addressContainer}
        />
      )}
      <Label type="regular" style={[styles.textColor, !comments && styles.detail]}>
        {t('visitDetails')}
      </Label>
      <View style={styles.detailContainer}>
        <View style={styles.content}>
          <Text type="small" textType="semiBold" style={textStyle}>
            {DateUtils.getDisplayDate(dateTime[0], 'DD MMM')}
          </Text>
          <Icon
            name={icons.roundFilled}
            color={isMissedVisit ? theme.colors.error : theme.colors.darkTint3}
            size={8}
            style={styles.iconStyle}
          />
          <Text type="small" textType="semiBold" style={textStyle}>
            {time?.formatted}
          </Text>
        </View>
        <TouchableOpacity style={styles.content} onPress={onPressSchedule}>
          <Icon name={icons.schedule} color={theme.colors.blue} size={20} />
          <Text type="small" style={styles.scheduleText}>
            {isCompletedVisit ? t('newVisit') : isRescheduleAll ? t('rescheduleAll') : t('reschedule')}
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  subAddress: {
    marginLeft: 0,
  },
  addressContainer: {
    marginVertical: 12,
  },
  textColor: {
    color: theme.colors.darkTint3,
  },
  detail: {
    marginTop: 12,
  },
  missedColor: {
    color: theme.colors.error,
  },
  detailContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconStyle: {
    marginTop: 6,
    marginHorizontal: 6,
  },
  scheduleText: {
    color: theme.colors.blue,
    marginStart: 6,
  },
});
