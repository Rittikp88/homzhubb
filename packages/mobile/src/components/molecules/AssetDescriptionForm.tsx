import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useTranslation } from 'react-i18next';
import { FormikProps, FormikValues } from 'formik';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { theme } from '@homzhub/common/src/styles/theme';
import { FormCalendar, FormDropdown, FormTextInput, Text } from '@homzhub/common/src/components';
import { FormCounter } from '@homzhub/common/src/components/molecules/FormCounter';
import { AssetDescriptionDropdownValues } from '@homzhub/common/src/domain/models/AssetDescriptionForm';

interface IProps {
  formProps: FormikProps<FormikValues>;
  dropDownOptions: AssetDescriptionDropdownValues;
}

const AssetDescriptionForm = ({ formProps, dropDownOptions }: IProps): React.ReactElement => {
  const [t] = useTranslation(LocaleConstants.namespacesKey.property);
  // Todo (Sriram: Look into this prefill issue of carpetArea)

  return (
    <>
      <Text type="small" style={styles.headingStyle}>
        {t('assetDescription:description')}
      </Text>
      <View style={styles.formContainer}>
        <View style={styles.contentView}>
          <View style={styles.subContentView}>
            <FormTextInput
              style={styles.inputFieldStyle}
              name="carpetArea"
              label={t('carpetArea')}
              maxLength={10}
              numberOfLines={1}
              inputType="number"
              placeholder={t('common:enter')}
              formProps={formProps}
            />
          </View>
          <View style={styles.flexOne}>
            <FormDropdown
              name="areaUnit"
              label={t('areaUnit')}
              options={dropDownOptions.areaUnitDropdownValues}
              placeholder={t('selectAreaUnit')}
              maxLabelLength={8}
              formProps={formProps}
            />
          </View>
        </View>
        <FormDropdown
          label={t('facingText')}
          name="facing"
          options={dropDownOptions.facing}
          placeholder={t('propertySearch:selectFacing')}
          maxLabelLength={36}
          formProps={formProps}
        />
        <FormDropdown
          label={t('flooringType')}
          name="flooringType"
          options={dropDownOptions.typeOfFlooring}
          placeholder={t('selectFlooringType')}
          maxLabelLength={36}
          formProps={formProps}
        />
        <FormCalendar
          formProps={formProps}
          label={t('yearOfConstruction')}
          name="yearOfConstruction"
          allowPastDates
          isYearView
          placeHolder={t('assetFinancial:addDatePlaceholder')}
          placeHolderStyle={styles.placeHolderStyle}
          dateStyle={styles.dateStyle}
          textType="label"
        />
        <FormCounter
          containerStyles={styles.marginTop}
          name="totalFloors"
          maxCount={1000}
          label={t('numberOfFloorsText')}
          formProps={formProps}
        />
        <FormCounter
          containerStyles={styles.marginTop}
          name="onFloorNumber"
          label={t('onFloorText')}
          formProps={formProps}
          maxCount={formProps.values.totalFloors}
        />
      </View>
    </>
  );
};

const memoizedComponent = React.memo(AssetDescriptionForm);
export { memoizedComponent as AssetDescriptionForm };

const styles = StyleSheet.create({
  headingStyle: {
    marginTop: 16,
    paddingVertical: 16,
    paddingLeft: 16,
    backgroundColor: theme.colors.moreSeparator,
  },
  formContainer: {
    backgroundColor: theme.colors.white,
    paddingHorizontal: 24,
    paddingBottom: 24,
  },
  contentView: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  subContentView: {
    flex: 1,
    marginRight: 16,
  },
  flexOne: {
    flex: 1,
  },
  marginTop: {
    marginTop: 24,
  },
  inputFieldStyle: {
    height: 40,
  },
  dateStyle: {
    marginLeft: 0,
  },
  placeHolderStyle: {
    color: theme.colors.darkTint8,
  },
});
