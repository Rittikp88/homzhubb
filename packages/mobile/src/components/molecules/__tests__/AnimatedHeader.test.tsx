import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AnimatedHeader } from '@homzhub/mobile/src/components/molecules/AnimatedHeader';

jest.mock('@homzhub/common/src/services/storage/StorageService', () => 'StorageService');
jest.mock('@react-native-community/google-signin', () => {});

describe('Test cases for Header', () => {
  const wrapper: ShallowWrapper = shallow(<AnimatedHeader />);

  it('should render snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
