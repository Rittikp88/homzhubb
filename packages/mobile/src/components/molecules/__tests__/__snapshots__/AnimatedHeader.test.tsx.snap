// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for Header should render snapshot 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ParallaxScrollView
    backgroundColor="#FFFFFF"
    backgroundScrollSpeed={5}
    backgroundSpeed={10}
    contentBackgroundColor="#fff"
    contentContainerStyle={null}
    fadeOutForeground={true}
    keyboardShouldPersistTaps="never"
    onChangeHeaderVisibility={[Function]}
    outputScaleValue={5}
    parallaxHeaderHeight={200}
    renderBackground={[Function]}
    renderContentBackground={[Function]}
    renderFixedHeader={[Function]}
    renderForeground={[Function]}
    renderParallaxHeader={[Function]}
    renderScrollComponent={[Function]}
    renderStickyHeader={[Function]}
    stickyHeaderHeight={100}
  />
</KeyboardAvoidingView>
`;
